lib/msf/core/exploit/remote/ms_samr.rb

```diff
@@ -108,9 +108,11 @@ def add_computer(opts = {})
   rescue RubySMB::Dcerpc::Error::SamrError => e
     raise MsSamrUnknownError, "A DCERPC SAMR error occured: #{e.message}"
   ensure
-    samr_con.samr.close_handle(user_handle) if user_handle
-    samr_con.samr.close_handle(samr_con.domain_handle) if samr_con.domain_handle
-    samr_con.samr.close_handle(samr_con.server_handle) if samr_con.server_handle
+    if samr_con
```

Comment:
This bonus fix addresses a stack trace in the samr_computer module that would occur if the `SMBDomain` argument was completely invalid.

Before:

```
msf6 auxiliary(admin/dcerpc/samr_computer) > show options 

Module options (auxiliary/admin/dcerpc/samr_computer):

   Name               Current Setting  Required  Description
   ----               ---------------  --------  -----------
   COMPUTER_NAME                       no        The computer name
   COMPUTER_PASSWORD                   no        The password for the new computer
   RHOSTS             192.168.159.10   yes       The target host(s), see https://github.com/rapid7/metasploit-framework/wiki/Using-Metasploit
   RPORT              445              yes       The target port (TCP)
   SMBDomain          foo.foo          no        The Windows domain to use for authentication
   SMBPass            Password1!       no        The password for the specified username
   SMBUser            smcintyre        no        The username to authenticate as


Auxiliary action:

   Name          Description
   ----          -----------
   ADD_COMPUTER  Add a computer account



View the full module info with the info, or info -d command.

msf6 auxiliary(admin/dcerpc/samr_computer) > run
[*] Running module against 192.168.159.10

[-] 192.168.159.10:445 - Auxiliary failed: NoMethodError undefined method `domain_handle' for nil:NilClass
[-] 192.168.159.10:445 - Call stack:
[-] 192.168.159.10:445 -   /home/smcintyre/Repositories/metasploit-framework/lib/msf/core/exploit/remote/ms_samr.rb:112:in `ensure in add_computer'
[-] 192.168.159.10:445 -   /home/smcintyre/Repositories/metasploit-framework/lib/msf/core/exploit/remote/ms_samr.rb:113:in `add_computer'
[-] 192.168.159.10:445 -   /home/smcintyre/Repositories/metasploit-framework/modules/auxiliary/admin/dcerpc/samr_computer.rb:70:in `action_add_computer'
[-] 192.168.159.10:445 -   /home/smcintyre/Repositories/metasploit-framework/modules/auxiliary/admin/dcerpc/samr_computer.rb:54:in `run'
[*] Auxiliary module execution completed
msf6 auxiliary(admin/dcerpc/samr_computer) > run SMBDomain=msflab
[*] Running module against 192.168.159.10

[+] 192.168.159.10:445 - Successfully created msflab\DESKTOP-KATWN3WT$
[+] 192.168.159.10:445 -   Password: wXHEsDw2ohkFnhudb5RJFVJ6wYfMZMbX
[+] 192.168.159.10:445 -   SID:      S-1-5-21-3402587289-1488798532-3618296993-2121
[*] Auxiliary module execution completed
msf6 auxiliary(admin/dcerpc/samr_computer) > 
```

After:
```
msf6 auxiliary(admin/dcerpc/samr_computer) > run
[*] Running module against 192.168.159.10

[-] 192.168.159.10:445 - Auxiliary aborted due to failure: unexpected-reply: Error returned during domain lookup in SAM server: (0xc00000df) STATUS_NO_SUCH_DOMAIN: The specified domain did not exist.
[*] Auxiliary module execution completed
msf6 auxiliary(admin/dcerpc/samr_computer) >
```